test/RachisTests/DatabaseCluster/AtomicClusterReadWriteTests.cs

```diff
@@ -327,10 +332,65 @@ public async Task ClusterWideTransaction_WhenRestoreFromIncrementalBackupAfterSt
                 return await session.LoadAsync<TestObj>(notDelete);
             });
 
-            var r = await AssertWaitForSingleAsync(async () => await documentStore.Operations.SendAsync(new GetCompareExchangeValuesOperation<TestObj>("")));
+            var r = await WaitForSingleAsync(async () => await documentStore.Operations.SendAsync(new GetCompareExchangeValuesOperation<TestObj>("")),timeout:500);
+            if (r.Count != 1)
+            {
+                // temp loggin to solve issue RavenDB-17890.
+                var logs = await socket.CloseAndGetLogsAsync();
+                Assert.True(false, $"Count is {r.Count} ,logs={logs}");
+            }
+
             Assert.EndsWith(notDelete, r.Single().Key, StringComparison.OrdinalIgnoreCase);
         }
 
+        private class DummyWebSocket : WebSocket
```

Comment:
move this to Tests.Infrastructure.Utils and remove other occurrences of this

Reply:
Done.